cpp/src/hash/concurrent_unordered_map.cuh

```diff
@@ -149,18 +157,53 @@ class concurrent_unordered_map {
         deleter};
   }
 
+  /**
+   * @note If called on host using managed allocator, then caller should ensure
```

Comment:
Add @brief. Also, this is a device-only function (unlike multi map), not sure why.

```suggestion
   * @brief Returns an iterator to the first element in the map
   *
```